lib/plug/conn.ex

```diff
@@ -415,23 +416,35 @@ defmodule Plug.Conn do
       Plug.Conn.send_file(conn, 200, "README.md")
 
   """
-  @spec send_file(t, status, filename :: binary, offset ::integer, length :: integer | :all) :: t | no_return
-  def send_file(conn, status, file, offset \\ 0, length  \\ :all)
+  @spec send_file(t, status, filename :: binary, offset :: integer, length :: integer | :all) ::
+          t | no_return
+  def send_file(conn, status, file, offset \\ 0, length \\ :all)
 
   def send_file(%Conn{state: state}, status, _file, _offset, _length)
-      when not(state in @unsent) do
+      when not (state in @unsent) do
     _ = Plug.Conn.Status.code(status)
     raise AlreadySentError
   end
 
-  def send_file(%Conn{adapter: {adapter, payload}, owner: owner} = conn, status, file, offset, length) when is_binary(file) do
+  def send_file(
```

Comment:
Same here:

```elixir
def send_file(conn, status, file, offset, length) when is_binary(file) do
  %Conn{adapter: {adapter, payload}, owner: owner} = conn
  # ...
end
```

Reply:
:+1:

Reply:
I would just match on the `%Conn{}` in the head because of better error messages in Elixir v1.5.

Reply:
Yeah, I completely forgot that when copy/pasting the param 😅 I've also updated the rest of the comments to match on `%Conn{} = conn`